src/quic_solana.rs

```diff
@@ -47,10 +47,21 @@ pub struct ConnectionCacheIdentity {
     shared: Arc<Mutex<QuicSessionInner>>,
     identity_watcher: watch::Sender<Pubkey>,
     reactive_signer: watch::Sender<PubkeySigner>,
+    stop_transactions: watch::Sender<()>,
+    allow_reset: watch::Receiver<()>,
 }
 
 impl ConnectionCacheIdentity {
-    pub async fn update_keypair(&self, keypair: &Keypair) {
+    pub async fn update_keypair(&self, keypair: &Keypair, is_reset: bool) {
+        let kp = keypair.insecure_clone();
+        let ps = PubkeySigner::new(kp);
+        self.reactive_signer.send_replace(ps);
+        if is_reset {
```

Comment:
do we meed is_reset here? can't this just be unaware of it and just you pass a random identity as keypair to this method?